tests/test_graphics_mem_safety.py

```diff
@@ -0,0 +1,65 @@
+"""Note: This tests are potential crashers (may result in segfaults)"""
+import pytest
+import imgui
+
+@pytest.fixture
+def context():
+    return imgui.create_context()
+
+@pytest.fixture
+def io():
+    # setup io
+    io = imgui.get_io()
+    io.delta_time = 1.0 / 60.0
+    io.display_size = 300, 300
+
+    # setup default font
+    io.fonts.get_tex_data_as_rgba32()
+    io.fonts.add_font_default()
+    io.fonts.texture_id = 42  # set any texture ID to avoid segfaults
+
+    return io
+
+def test_texture_id_int_reference(context, io):
+
+    # See issue #248 (https://github.com/pyimgui/pyimgui/issues/248)
+
+    texture_id = 0
+    for i in range(0,10000):
+        imgui.new_frame()
+
+        imgui.begin("tests")
+        io.fonts.texture_id = texture_id
+        imgui.image(texture_id, 640, 480)
+        imgui.image_button(texture_id, 200, 50)
+        draw_list = imgui.get_background_draw_list()
+        draw_list.add_image(texture_id, (20, 35), (180, 80))
+        imgui.end()
+
+        texture_id += 1
+
+        imgui.render()
+        
+        draw_data = imgui.get_draw_data()
+        for commands in draw_data.commands_lists:
+            for command in commands.commands:
+                if type(command.texture_id) is imgui.core._DrawList:
```

Comment:
Not sure if you want this check here?
texture_id should be nothing or an `int`, right?

Reply:
Instead, i'd skip only specifically `io.fonts.texture_id` here, just like in the test below.

Reply:
> texture_id should be nothing or an int, right?

Actually no, `texture_id` can be anything and imgui does not make any assumption on the type of texture id. It is an int for OpenGL but it could be a pointer to an object that represents the texture or anything that is needed. This is why we did go for the cache of references instead of forcing it to be an integer as previously proposed in #249.

I had to add that check because using draw lists was leading to a special case where the first `texture_id` returned was the draw list itself. However, it clearly looks fishy now with some afterthought. I'll look into that.

Reply:
`io.fonts.texture_id` is passed as an integer in this example and uses the same texture index as the other call, thus it shall be tested too. The second test only checks that we get back what we gave in, which is not necessarily an int.

Reply:
Right, but I would say that this check could be hiding a bug,
since in the 'bad' behavior this is exactly what happened: an int was replaced with a drawlist.

Reply:
Indeed. I don't know what happened, probably a wrong manipulation from my side. Anyway, I removed the test as you suggested and everything seems to work fine without it. Thanks for the review!